### PR TITLE
201 continous deplyoment

### PR DESCRIPTION
**Pull request by GiorgioCerro**
This PR introduces ESLint, the linting for JS, with a plugin specifically for TS and new rules for prettier. 

New dependencies: 
- eslint v8.56.0
- prettier v3.1.0
- typescript-eslint

At the moment, in the .github/workflow/pre-commit.yaml there is an extra line of code to install the dependencies with npm.

There are three config files to look at in dtbase/webapp:
- .eslintrc, where we can set the rules for eslint and which rules to apply
- .prettierrc, the rules for prettier

It is better to not replicate rules in both the config files, others there is a possibility to encounter conflicts between the two linters.

Prettier fixes these issues:
{
    "tabWidth": 2,  # tab size
    "semi": false,  # whether to put semicolons at the end of every line
    "printWidth": 88, 
    "singleQuote": false # to use only single quotes
}

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -20,8 +20,15 @@ repos:
...
   hooks:
   - id: prettier
     files: ^dtbase/webapp
     args: [--ignore-path=dtbase/webapp/.prettierignore]
+- repo: https://github.com/pre-commit/mirrors-eslint
+  rev: v8.56.0
+  hooks:
+  - id: eslint
+    exclude: ^dtbase/webapp/node_modules
+    files: \.tsx?$
```
Comment (mhauru) [inline .pre-commit-config.yaml:33]: Does eslint also do .css files? If yes, could we include those?

File: .github/workflows/pre-commit.yaml [hunk 2]
```diff
@@ -16,4 +16,5 @@ jobs:
         python -m ensurepip
         python -m pip install --upgrade pip
         python -m pip install .[dev]
+        npm install --prefix dtbase/webapp/ --include=dev
```
Comment (mhauru) [inline .github/workflows/pre-commit.yaml:19]: Is this only needed for the `eslint-plugin-prettier` thing, or does something else require this too? If it's only the plugin, do we really need it?

Comment (mhauru) [inline .github/workflows/pre-commit.yaml:19]: Also, if this is needed, we should probably add a similar note to the developer documentation, at the same point when we advice people how to use pre-commit. (I'm assuming we say something about pre-commit in the dev docs.)

**Review by mhauru — COMMENTED** [review]
Looks great, just had a couple of questions about the plugin/docs and css files.

File: .github/workflows/pre-commit.yaml (hunk 2 above)
Comment (GiorgioCerro) [reply .github/workflows/pre-commit.yaml]: I agree, though do you want me to do this right now or leave it open for later when you have finished with the other PR? I am not fully aware of how you are restructuring all the documentation

Comment (mhauru) [reply .github/workflows/pre-commit.yaml]: If you want to add a comment in the dev docs feel free to go ahead, I can handle merging it with the docs rewrite. If you are busy we can merge this as is and I'll add the comment later.

Comment (GiorgioCerro) [reply .github/workflows/pre-commit.yaml]: Sure, I will do that

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mhauru) [reply .pre-commit-config.yaml:33]: Happy with that, thanks for checking.

**Review by mhauru — APPROVED** [review]
Thanks Giorgio, great work!

**Pull request merged**